### PR TITLE
Document urltestdata.json

### PR DESCRIPTION
**Pull request by SimonSapin**

### CODE REVIEW

**Comment by zcorpan** [comment]
Looks OK to me from a quick look but I suppose @domenic or @annevk should sanity-check also.

File: url/README.md [hunk 1]
```diff
@@ -1,6 +1,19 @@
-The test for browsers is `a-element.html`. The reusable format is `urltestdata.txt`, which
-is not documented in detail. Reverse engineering through `urltestparser.js` should not be
-too hard. Documentation welcome!
+The test for browsers is `a-element.html`. The reusable format is `urltestdata.json`,
+which contains a JSON array of comments as strings and test cases as objects.
+The keys for each test case are:
+
+* `base`: an absolute URL as a string whose [parsing] without a base of its own should succeed.
```
Comment (domenic) [inline url/README.md]: Maybe document that URLs are always parsed with a base; "absolute URLs" are parsed against "about:blank"

**Comment by domenic** [comment]
LGTM with minor suggestion

File: url/README.md (hunk 1 above)
Comment (domenic) [inline url/README.md]: The tests expect them, I meant.

Comment (SimonSapin) [inline url/README.md]: Oh, I see. How about adding this?

> This key is always present, and may have a value like `"about:blank"` when `input` is an absolute URL.

Comment (domenic) [inline url/README.md]: Yeah that makes sense.

Comment (SimonSapin) [inline url/README.md]: Done.

File: url/README.md [hunk 2]
```diff
@@ -1,6 +1,21 @@
-The test for browsers is `a-element.html`. The reusable format is `urltestdata.txt`, which
-is not documented in detail. Reverse engineering through `urltestparser.js` should not be
-too hard. Documentation welcome!
+The test for browsers is `a-element.html`. The reusable format is `urltestdata.json`,
```
Comment (domenic) [inline url/README.md]: While you're here maybe note that the tests are actually:

- a-element.html
- url-constructor.html
- a-element-xhtml.xhtml (although this seems like I forgot to update it :()

Comment (SimonSapin) [inline url/README.md]: Done.

**Comment by domenic** [comment]
Travis is failing because of trailing whitespace in README.md. LGTM after Travis becomes happy.

**Comment by SimonSapin** [comment]
I blame the github editor for not showing whitespace :) Fixed.

**Comment by domenic** [comment]
Merged as c54821c95bdf71edf28f2a818e8a7d372d994a77, thanks!!

**Pull request closed without merging**